src/client/datascience/kernel-launcher/kernelLauncherDaemon.ts

```diff
@@ -51,14 +51,20 @@ export class PythonKernelLauncherDaemon implements IDisposable {
         interpreter?: PythonEnvironment
     ): Promise<{ observableOutput: ObservableExecutionResult<string>; daemon: IPythonKernelDaemon | undefined }> {
         traceInfo(`Launching kernel daemon for ${kernelSpec.display_name} # ${interpreter?.path}`);
-        const [daemon, wdExists, env] = await Promise.all([
+        let [daemon, wdExists, env] = await Promise.all([
             this.daemonPool.get(resource, kernelSpec, interpreter),
             fs.pathExists(workingDirectory),
             this.kernelEnvVarsService.getEnvironmentVariables(resource, interpreter, kernelSpec)
         ]);
 
         // Check to see if we have the type of kernelspec that we expect
         const args = kernelSpec.argv.slice();
+        if (resource && resource.fsPath.toLowerCase().endsWith('.py')) {
```

Comment:
Without the change to KernelProvider, this resource will be the Ingeractive Notebook Uri when using native itneractive window (ie. this used to work only in webview interactive, now works everywhere)

Reply:
THis isn't going to work when the interactive window is set to 'single' or is multiple and you're sharing the IW amongst different files. 

Only the first python file will be debuggable.

Reply:
I think we need to change the env value before every run of the cell if debugging.

Reply:
We could likely do that in the cellHashProvider in the preExecute function. It could either add a line to the cell to do it or execute silently before the execute. The extra line would just make the offset for the breakpoint one larger.

Reply:
Oh I guess putting it in the cell doesn't work because it needs it before it runs the cell. So it would need to execute some code ahead of time.

Reply:
I'll try execution of the code silently.

Reply:
Agreed, done.